AutomatedTesting/Gem/PythonTests/largeworlds/landscape_canvas/EditorScripts/ComponentUpdates_UpdateGraph.py

```diff
@@ -90,113 +90,83 @@ def ComponentUpdates_UpdateGraph():
-    Report.critical_result(Tests.lc_entity_found, prefab_root_id.IsValid())
+    prefab_root_entity = EditorEntity.find_editor_entity("LandscapeCanvas")
+    Report.critical_result(Tests.lc_entity_found, prefab_root_entity.id.IsValid())
 
     # Find the BushSpawner entity
-    search_filter.names = ["BushSpawner"]
-    spawner_matching_entities = entity.SearchBus(bus.Broadcast, 'SearchEntities', search_filter)
-    spawner_id = spawner_matching_entities[0]       #Entity with Vegetation Layer Spawner component
-    Report.critical_result(Tests.spawner_entity_found, spawner_id.IsValid())
-
-    # Get needed component type ids
-    distribution_filter_type_id = hydra.get_component_type_id("Vegetation Distribution Filter")
-    altitude_filter_type_id = hydra.get_component_type_id("Vegetation Altitude Filter")
-    box_shape_type_id = hydra.get_component_type_id("Box Shape")
+    bush_spawner_entity = EditorEntity.find_editor_entity("BushSpawner")
+    Report.critical_result(Tests.spawner_entity_found, bush_spawner_entity.id.IsValid())
 
     # Verify the BushSpawner entity has a Distribution Filter
-    has_distribution_filter = editor.EditorComponentAPIBus(bus.Broadcast, 'HasComponentOfType', spawner_id,
-                                                           distribution_filter_type_id)
+    has_distribution_filter = bush_spawner_entity.has_component("Vegetation Distribution Filter")
     Report.critical_result(Tests.dist_filter_component_found, has_distribution_filter)
 
     # Open Landscape Canvas and the existing graph
-    general.open_pane('Landscape Canvas')
-    open_graph = landscapecanvas.LandscapeCanvasRequestBus(bus.Broadcast, 'OnGraphEntity', prefab_root_id)
+    general.open_pane("Landscape Canvas")
+    open_graph = landscapecanvas.LandscapeCanvasRequestBus(bus.Broadcast, "OnGraphEntity", prefab_root_entity.id)
     Report.critical_result(Tests.existing_graph_opened, open_graph.IsValid())
 
     # Verify that Distribution Filter node is present on the graph
-    spawner_distribution_filter_component = editor.EditorComponentAPIBus(bus.Broadcast, 'GetComponentOfType', spawner_id,
-                                                                         distribution_filter_type_id)
-    spawner_distribution_filter_component_id = spawner_distribution_filter_component.GetValue()
-    distribution_filter_node = landscapecanvas.LandscapeCanvasRequestBus(bus.Broadcast,
-                                                                         'GetNodeMatchingEntityComponentInGraph',
-                                                                         open_graph,
-                                                                         spawner_distribution_filter_component_id)
-    Report.critical_result(Tests.dist_filter_node_found, distribution_filter_node is not None)
+    spawner_dist_filter_component = bush_spawner_entity.get_components_of_type(["Vegetation Distribution Filter"])[0]
```

Comment:
Does the size of the list need to get verified before dereferencing the first element?

Reply:
If no entity/component pairs are found, get_components_of_type would assert and interrupt the test.